web_widget_uom/static/src/js/widget.js

```diff
@@ -0,0 +1,48 @@
+/* Global jscolor */
+odoo.define("web.web_widget_uom", function (require) {
+    "use strict";
+
+    var basic_fields = require("web.basic_fields");
+    var field_registry = require("web.field_registry");
+
+    var FieldUoM = basic_fields.FieldFloat.extend({
+        /**
+         * Main rendering function.Override this if your widget has the same render
+         * for each mode.  Note that this function is supposed to be idempotent:
+         * the result of calling 'render' twice is the same as calling it once.
+         * Also, the user experience will be better if your rendering function is
+         * synchronous.
+         *
+         * @private
+         */
+        _render: function () {
+            var self = this;
+            var parent = this._super;
+            Promise.resolve(this._getDecimalPlaces()).then(function () {
```

Comment:
Can you move this to`willStart` that method is more appropriate for this kind of task.

Reply:
done

Reply:
`willStart` exists in legacy `web.Widget`.

Something like this:
```
    willStart: function () {
        return Promise.all([
            this._super.apply(this, arguments),
            this._getDecimalPlaces()
        ]);
```

Reply:
Done, thx.